apps/web/components/dashboard/BulkBookmarksAction.tsx

```diff
@@ -0,0 +1,204 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { usePathname } from "next/navigation";
+import { Button, ButtonWithTooltip } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { useToast } from "@/components/ui/use-toast";
+import bulkActions from "@/store/bulkBookmarksAction";
+import { Archive, ArchiveRestore, Pencil, Star, Trash2, X } from "lucide-react";
+
+import {
+  useDeleteBookmark,
+  useUpdateBookmark,
+} from "@hoarder/shared-react/hooks/bookmarks";
+
+function DeleteModal({
+  open,
+  setOpen,
+  deleteBookmarks,
+}: {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  deleteBookmarks: () => void;
+}) {
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
```

Comment:
Sounds like you can use the `ActionConfirmingDialog` here.

Reply:
Resolved